libs/did-resolver-lib/src/resolver.ts

```diff
@@ -0,0 +1,125 @@
+import { DIDDoc, DIDResolver, Service, VerificationMethod } from 'didcomm';
+import base64url from 'base64url';
+
+type Purpose =
+  | 'Assertion'
+  | 'Encryption'
+  | 'Verification'
+  | 'CapabilityDelegation'
+  | 'CapabilityInvocation'
+  | 'Service';
+
+export default class PeerDIDResolver implements DIDResolver {
+  async resolve(did: string): Promise<DIDDoc | null> {
+    try {
+      // Validate if the DID starts with the "did:peer:" prefix
+      if (!did.startsWith('did:peer:')) {
+        throw new Error('Unsupported DID method');
+      }
+
+      // Dissect the DID address
+      const chain = did
+        .replace(/^did:peer:2\./, '')
+        .split('.')
+        .map((e) => {
+          const purposeCode = e.charAt(0);
+          const purpose = mapPurposeFromCode(purposeCode);
+          const multikey = e.slice(1);
+          return { purpose, multikey };
+        });
+
+      const authentication: string[] = [];
+      const keyAgreement: string[] = [];
+      const assertionMethod: string[] = [];
+      const verificationMethods: VerificationMethod[] = [];
+
+      chain
+        .filter(({ purpose }) => purpose !== 'Service')
+        .forEach((item, index) => {
+          const id = `#key-${index + 1}`;
+          const { purpose, multikey } = item;
+
+          switch (purpose) {
+            case 'Assertion':
+              assertionMethod.push(id);
+              break;
+            case 'Verification':
+              authentication.push(id);
+              break;
+            case 'Encryption':
+              keyAgreement.push(id);
+              break;
+          }
+
+          const method: VerificationMethod = {
+            id,
+            type: 'Multikey',
+            controller: did,
+            publicKeyMultibase: `z${multikey}`,
+          };
+
+          verificationMethods.push(method);
+        });
+
+      // Resolve services
+      const services: Service[] = [];
+      let serviceNextId = 0;
+
+      chain
+        .filter(({ purpose }) => purpose === 'Service')
+        .forEach(({ multikey }) => {
+          const decodedService = base64url.decode(multikey);
+          const service = reverseAbbreviateService(decodedService);
+
+          if (!service.id) {
+            service.id =
+              serviceNextId === 0 ? '#didcomm' : `#didcomm-${serviceNextId}`;
```

Comment:
Please use `#service` here, instead of `#didcomm`.

![image](https://github.com/user-attachments/assets/aa7c45f6-e268-49bc-9de0-6e280df57209)

Reply:
okay